azext_edge/edge/providers/rpsaas/adr/base.py

```diff
@@ -0,0 +1,210 @@
+            logger.warning(CUSTOM_LOCATION_NOT_FOUND_ERROR.format(custom_location_id))
+            return
+        query = f'| where id =~ "{custom_location_query_result[0]["properties"]["hostResourceId"]}"'
+        cluster_query_result = build_query(
+            self.cmd,
+            custom_query=query,
+            type=ResourceTypeMapping.connected_cluster.value
+        )
+        # TODO: add option to fail on these
+        if len(cluster_query_result) == 0:
+            logger.warning(CLUSTER_NOT_FOUND_ERROR.format(custom_location_id))
+            return
+        cluster = cluster_query_result[0]
+        if cluster["properties"]["connectivityStatus"].lower() != "connected":
+            logger.warning(CLUSTER_OFFLINE.format(cluster["name"]))
+
+    def _check_cluster_and_custom_location(
+        self,
+        custom_location_name: Optional[str] = None,
+        custom_location_resource_group: Optional[str] = None,
+        custom_location_subscription: Optional[str] = None,
+        cluster_name: Optional[str] = None,
+        cluster_resource_group: Optional[str] = None,
+        cluster_subscription: Optional[str] = None,
+    ) -> Dict[str, str]:
+        if not any([cluster_name, custom_location_name]):
+            raise RequiredArgumentMissingError(MISSING_CLUSTER_CUSTOM_LOCATION_ERROR)
+        query = ""
+        cluster = None
+
+        # provide cluster name - start with checking for the cluster (if can)
+        if cluster_name:
+            cluster_query_result = build_query(
+                self.cmd,
+                subscription_id=cluster_subscription,
+                type=ResourceTypeMapping.connected_cluster.value,
+                name=cluster_name,
+                resource_group=cluster_resource_group
+            )
+            if len(cluster_query_result) == 0:
+                raise ResourceNotFoundError(f"Cluster {cluster_name} not found.")
+            if len(cluster_query_result) > 1:
+                raise ValidationError(
+                    MULTIPLE_POSSIBLE_ITEMS_ERROR.format(len(cluster_query_result), "cluster", cluster_name)
+                )
+            cluster = cluster_query_result[0]
+            # reset query so the location query will ensure that the cluster is associated
+            query = f"| where properties.hostResourceId =~ \"{cluster['id']}\" "
+
+        # try to find location, either from given param and/or from cluster
+        # if only location is provided, will look just by location name
+        # if both cluster name and location are provided, should also include cluster id to narrow association
+        location_query_result = build_query(
+            self.cmd,
+            subscription_id=custom_location_subscription,
+            custom_query=query,
+            type=ResourceTypeMapping.custom_location.value,
+            name=custom_location_name,
+            resource_group=custom_location_resource_group
+        )
+        if len(location_query_result) == 0:
```

Comment:
Is there symantic value for using `len()`? Why not `if not location_query_result:`?

Reply:
Same question here 
![image](https://github.com/Azure/azure-iot-ops-cli-extension/assets/28658112/ef36a03c-5e81-4f07-a932-045057819171)

Reply:
... very yes